packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -236,6 +242,90 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+- (void)setPictureInPicture:(BOOL)pictureInPictureEnabled {
+  if (self.pictureInPictureEnabled == pictureInPictureEnabled) {
+    return;
+  }
+
+  self.pictureInPictureEnabled = pictureInPictureEnabled;
+  if (_pipController && self.pictureInPictureEnabled &&
+      ![_pipController isPictureInPictureActive]) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"startingPiP"});
+    }
+    [_pipController startPictureInPicture];
+  } else if (_pipController && !self.pictureInPictureEnabled &&
+             [_pipController isPictureInPictureActive]) {
+    [_pipController stopPictureInPicture];
+  }
+}
+
+#if TARGET_OS_IOS
+- (void)setupPipController:(BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if ([AVPictureInPictureController isPictureInPictureSupported]) {
+    _pipController = [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+    if (@available(iOS 14.2, *)) {
+      _pipController.canStartPictureInPictureAutomaticallyFromInline =
+          canStartPictureInPictureAutomaticallyFromInline;
+    }
+    _pipController.delegate = self;
+  }
+}
+
+- (void)usePlayerLayer:(CGRect)frame
+    canStartPictureInPictureAutomaticallyFromInline:
+        (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (_player) {
+    self.playerLayer = [AVPlayerLayer playerLayerWithPlayer:_player];
+    UIViewController *vc = [[[UIApplication sharedApplication] keyWindow] rootViewController];
+    self.playerLayer.frame = frame;
+    self.playerLayer.needsDisplayOnBoundsChange = YES;
+    self.playerLayer.opacity = 0.001;
```

Comment:
what does 0.001 opacity mean?

Reply:
Because the player is an overlay and flutter is drawn behind the overlay we need to set the opacity to something very low. Native iOS PiP UI, when switching between tabs. (video in futter will change location, but the location of the playerLayer is kept the same) By setting the opacity to 0.001 you don't have this issue. 

This is a hacky solution. But we couldn't find another way to fix this.

Reply:
It's hard to tell from reading the code. can you add a comment there.

Reply:
I added extra comments.